src/components/ebay-carousel/index.js

```diff
@@ -286,37 +286,27 @@ function handleScrollEnd(scrollLeft) {
     }
 
     const { state } = this;
-    const { config, items } = state;
-
-    // Find the closest item using a binary search.
-    let start = 0;
-    let end = items.length - 1;
-    let remaining;
-    let closest;
-
-    while (end - start > 1) {
-        remaining = end - start;
-        const middle = start + Math.floor(remaining / 2);
-        if (scrollLeft < items[middle].left) end = middle;
-        else start = middle;
-    }
-
-    if (remaining === 0) {
-        closest = start;
-    } else {
-        const deltaStart = Math.abs(scrollLeft - items[start].left);
-        const deltaEnd = Math.abs(scrollLeft - items[end].left);
-        closest = deltaStart < deltaEnd ? start : end;
-    }
+    const { config, items, slideWidth } = state;
+    const itemsPerSlide = state.itemsPerSlide || 1;
+    const direction = getOffset(state) > scrollLeft ? RIGHT : LEFT;
+    // Used to add additional tolerance based on swipe direction.
+    const targetLeft = scrollLeft - (direction * slideWidth * 0.2);
 
-    const closestOffset = items[closest].left;
-    const maxOffset = getMaxOffset(state);
+    // Find the closest item using a binary search on each carousel slide.
+    const totalItems = items.length;
+    let low = 0;
+    let high = Math.ceil(totalItems / itemsPerSlide) - 1;
 
-    // If we are closer to the end than the closest item, then we just go to the end.
-    if (Math.abs(maxOffset - scrollLeft) < Math.abs(closestOffset - scrollLeft)) {
-        closest = items.length - 1;
+    while (high - low > 1) {
+        const mid = Math.floor((low + high) / 2);
+        if (targetLeft > items[mid * itemsPerSlide].left) low = mid;
+        else high = mid;
```

Comment:
I know this is shorthand, but eBay's code style convention is to use normal `if` blocks. I know that kinda sucks, but ... I'd like to keep it consistent everywhere.

Reply:
This is how it was before, so we should probably lint for this. Either way I'll fix it.

Reply:
I know it's done now, but for future reference I'd prefer us not to touch unrelated lines of code (like syntax changes) in bug fixes. The "lines changed" in the PR is now significantly more than it should be. Sounds like nitpicking, but anything to help speed up PR reviews, and reduces future merge conflicts, is a godsend to me!!!

Anyhow, changes look good. 

Anyway, changes look good.

Reply:
I agree. Again the linter should catch this (or maybe there should be no stylistic rules ALA prettier 🙏). My only other options were to not address the PR comment (which was valid), or to only update a portion of the file which would make it inconsistent.

Reply:
When making the comment, I didn't realize there were others in the file. Oh well.

Reply:
Agree with @DylanPiercey, this is a linter's job. We have explicitly set the [`curly`](https://eslint.org/docs/rules/curly) linter rule to [allow](https://github.com/eBay/eslint-config-ebay/blob/96f55c9b3eafb4c5ac07e536dbfc271a2d1a55f4/rules/bp.js#L14) this option. This is part of [eslint-config-ebay](https://github.com/eBay/eslint-config-ebay), which every team is using. Maybe we should have a ebayui specific override if needed.

Reply:
@senthilp I didn't know that the linter allowed that! If that's the case, then I rescind my comment, and I apologize to @DylanPiercey for all this confusion.

Reply:
No worries. Personally, I like your suggestion. The rule should be like `"curly": [2, "all"]`. But the general consensus was to allow single statement `if` conditions to be without braces.